storage.go

```diff
@@ -3,34 +3,161 @@ package minio
 import (
 	"context"
 	"io"
+	"strings"
 
+	"github.com/minio/minio-go/v7"
+
+	ps "github.com/beyondstorage/go-storage/v4/pairs"
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
+	"github.com/beyondstorage/go-storage/v4/services"
 	. "github.com/beyondstorage/go-storage/v4/types"
 )
 
+const defaultListObjectBufferSize = 100
+
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	rp := s.getAbsPath(path)
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		if !s.features.VirtualDir {
+			return
+		}
+		rp += "/"
+		o = s.newObject(true)
+		o.Mode = ModeDir
+	} else {
+		o = s.newObject(false)
+		o.Mode = ModeRead
+	}
+	o.ID = rp
+	o.Path = path
+	return o
 }
 
 func (s *Storage) delete(ctx context.Context, path string, opt pairStorageDelete) (err error) {
-	panic("not implemented")
+	rp := s.getAbsPath(path)
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		if !s.features.VirtualDir {
+			err = services.PairUnsupportedError{Pair: ps.WithObjectMode(opt.ObjectMode)}
+			return
+		}
+		rp += "/"
+	}
+	err = s.client.RemoveObject(ctx, s.bucket, rp, minio.RemoveObjectOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 func (s *Storage) list(ctx context.Context, path string, opt pairStorageList) (oi *ObjectIterator, err error) {
-	panic("not implemented")
+	rp := s.getAbsPath(path)
+	options := minio.ListObjectsOptions{WithMetadata: true}
```

Comment:
We support [Object Lazy Stat](https://beyondstorage.io/docs/go-storage/internal/object-lazy-stat), which allow the user to get the information only when they really needed.

So we don't need `WithMetadata: true` here.